### ISSUE TITLE
CCC Output tables major groupings

### ISSUE THREAD

**Issue by jdebacker** [OpenSourcePolicyCenter/webapp-public#326]
Major groupings for industries and assets would be helpful since the names themselves are sometimes not that descriptive without such groupings.  

The idea would be to have the name of the major grouping offset - so it is left aligned and the more detailed industries/assets under that are indented.

**Comment by jdebacker** [comment]
To add to this further, the current alphabetical ordering of industries or assets makes the tables difficult to read.  Industries should be ordered in accordance with [NAICS code](http://www.census.gov/cgi-bin/sssd/naics/naicsrch?chart=2012) ordering, since people are familiar with that.  Major asset groups should be ordered: Equipment, Structures, IP, Land, Inventories.

When adding major groupings, we should change output tables from B-Tax to include average rates for the major groupings.

In addition, it'd be helpful to have an overall rate, across all assets/industries.

**Comment by talumbau** [comment]
@PeterDSteinberg can you comment here on ideas for how this grouping would be accomplished? I'm thinking in terms of the JSON key-value pairs we get back for rendering. Perhaps we could have some kind formatting value included with every row - a string that indicates the grouping for row? then we could organize the rows by group. What do you think?

**Comment by brendancol** [comment]
@Jason Do you have a screenshot of an analogous tables like what we'd want for B-Tax output?

**Comment by jdebacker** [comment]
@brendancol Here's an example of industry groupings:
![ind_ss](https://cloud.githubusercontent.com/assets/10715924/19490003/5dfa1048-953b-11e6-9dd8-36a6baf32fb9.png)

What you can take from this is that I think it's nice to have the major grouping left aligned (and perhaps in bold font) and then the group members indented.  

The output tables do not currently include the major groupings.  Let me know how  you'd like those tables formatted (e.g. major group identified in separate column or with some notation in the industry name) and I can do that.

I think we should have similarly formatted tables for "by asset", but there just have five groups: equipment, structures, intellectual property, inventories, and land.

Also, I have a particular ordering in mind for the industries and assets.  So let me know what needs to be done to the B-Tax output tables to ensure that we can order them in the web app.

**Comment by jdebacker** [comment]
@brendancol Something to think about - if we could make the tables expandable - e.g. just showing the major grouping and then the user clicks a "+" button to expand to the sub-industries, that'd be cool, but certainly not necessary.

**Comment by brendancol** [comment]
@Jason, great thanks. I'm going to come up with a proposal which retains use of `DataTables`.  Keeping `DataTables` will significantly reduce the level of effort.

**Comment by brendancol** [comment]
@jdebacker yeah go ahead and add any fields which would help with groupby.

**Comment by jdebacker** [comment]
@brendancol Ok - I'll update B-Tax output tables to have an additional column, "major_grouping", which will be a text field with the name of the major industry/asset grouping.  

I'm also going to include values for averages across the group.  In this case, "major_grouping" will equal the industry or asset name.

**Comment by brendancol** [comment]
@jdebacker sounds good.  Will `major_grouping` included aggregated values for all corporate/non-corporate categories?

**Comment by jdebacker** [comment]
@brendancol, yes, this is what I'm referring to in the last two sentences above. 

as an example:
```
major_grouping	asset	mettr_c	mettr_nc
equipment	equipment	0.33	0.27
equipment	computers	0.25	0.23
equipment	furniture	0.34	0.31
```

**Comment by brendancol** [comment]
@jdebacker is `major_grouping` the only groupby, or will there also be a `minor_grouping`?

**Comment by jdebacker** [comment]
@brendancol If you can handle it, I'd like to have a minor grouping for industry (but not asset).  e.g se the manufacturing breakout above (but most major groupings will only have one minor group).

I also would like a line for the overall averages for the variables (i.e., across all assets/industries).  For this, I thought it'd be put in the table as follows:

```
major_grouping  asset   mettr_c mettr_nc
overall     overall     0.35    0.33
equipment   equipment   0.33    0.27
equipment   computers   0.25    0.23
equipment   furniture   0.34    0.31
```

**Comment by brendancol** [comment]
@jdebacker will there be a mettr_c_reform, mettr_c_baseline etc.?

**Comment by jdebacker** [comment]
@brendancol All the columns of output will remain the same - just adding new columns for the major groupings.  Should have been clear - example was simple to just quickly show how the major groupings would work with the asset/industry names.

**Comment by brendancol** [comment]
@jdebacker @PeterDSteinberg @talumbau 

At the moment, the `btax/results.html` now supports items which have a `major_grouping` property which would be a sibling to `label` and `cells` in each of the objects.

**Comment by talumbau** [comment]
closed via #377
